src/sidebar/store/modules/selection.js

```diff
@@ -374,14 +375,15 @@ function setFocusModeFocused(focused) {
 }
 
 /**
- * Changes the focused user and sets focused enabled to `true`.
+ * Clears any applied filters, changes the focused user and sets
+ * focused enabled to `true`.
  *
  * @param {User} user - The user to focus on
  */
 function changeFocusModeUser(user) {
-  return {
-    type: actions.CHANGE_FOCUS_MODE_USER,
-    user,
+  return function (dispatch) {
```

Comment:
I did use a `thunk` here, but in a rather low-key way.

Reply:
An alternate possibility would be to handle both of these steps in the `CHANGE_FOCUS_MODE_USER` reducer. If you want to avoid repeating the `CLEAR_SELECTION` logic, you can make that logic a separate top-level function which the `CHANGE_FOCUS_MODE_USER` function just re-uses, eg.

```js
function clearSelectionReducer(state, action) {
...
}

const update = {
  CLEAR_SELECTION: clearSelectionReducer,
  CHANGE_FOCUS_MODE_USER(state, action) {
   let focusMode = ...
   return {
      ...clearSelectionReducer(state, action),
      focusMode,
   };
  }
}
```

The advantage of combining these actions is that a) it is a bit more efficient (only one action passes through the store and triggers updates, instead of two) and, importantly in some cases, b) it makes the update atomic to outside observers (ie. they will not see the intermediate state where only one of the two actions has been dispatched).

In this particular case, I suspect neither of these advantages is particularly important.